Resolves yortus/DBFFile#80

zoozalp opened the pull request.

yortus commented:
Thanks for the PR @zoozalp! In order to merge this, there would need to be a 'proper' solution - maybe disabling the length validation in loose mode only, and possibly only for certain file versions. I'd be curious if the dbf this PR is designed to fix is actually valid? Do you have the dbf version of the file?

The other thing would be to add at least one new unit test and a test fixture (i.e. a dbf file that currently fails but would pass with this change).

Also it would be good to only submit the actual change - most of this PR is formatting changes with no code changes - I'd revert all that and restore the original formatting. It's currently hard to spot the single line that has changed.

zoozalp commented:
It's a valid dbase 3 file I pulled from production. Since I don't have tools to edit it or create a sample dbf file, I can't take care of this this weekend but I'll try to do it next week.

And I see my IDE reformatted source code. I've reverted and redid it on my local. In case you're wondering, here are only changes I made;

1. I've added following code after line 144 of src/dbf-file.ts
`
            if (field.type === "C" && field.decimalPlaces) {
                field.size = field.decimalPlaces * 256 + field.size;
            }
`

2. disabled line 39 of src/file-descriptor.ts
`
    //if (type === 'C' && size > 255) throw new Error('Field size is too 
large (maximum is 255)');
`
